xchange-blockchain/src/main/java/org/knowm/xchange/blockchain/BlockchainAdapters.java

```diff
@@ -0,0 +1,256 @@
+                    .averagePrice(blockchainOrder.getAvgPx())
+                    .build();
+
+            if (order instanceof LimitOrder) {
+                limitOrders.add((LimitOrder) order);
+            } else {
+                hiddenOrders.add(order);
+            }
+        }
+
+        return new OpenOrders(limitOrders, hiddenOrders);
+    }
+
+    public static UserTrades toUserTrades(List<BlockchainOrder> blockchainTrades) {
+        List<UserTrade> trades = blockchainTrades.stream()
+                .map(blockchainTrade -> new UserTrade.Builder()
+                        .type(blockchainTrade.isBuyer()? Order.OrderType.BID : Order.OrderType.ASK)
+                        .originalAmount(blockchainTrade.getCumQty())
+                        .currencyPair(blockchainTrade.getSymbol())
+                        .price(blockchainTrade.getPrice())
+                        .timestamp(blockchainTrade.getTimestamp())
+                        .id(Long.toString(blockchainTrade.getExOrdId()))
+                        .orderId(blockchainTrade.getClOrdId())
+//                       .feeAmount(t.commission)
+//                        .feeCurrency(Currency.getInstance(t.commissionAsset))
+                        .build()
+                ).collect(Collectors.toList());
+        Long lastId = blockchainTrades.stream().map(BlockchainOrder::getExOrdId).max(Long::compareTo).orElse(0L);
+        return new UserTrades(trades, lastId, Trades.TradeSortType.SortByTimestamp);
+    }
+
+    public static BlockchainOrder toBlockchainOrder(LimitOrder limitOrder){
+        return BlockchainOrder.builder()
+                .ordType(LIMIT)
+                .symbol(toCurrencyPair(limitOrder.getInstrument()))
+                .side(Order.OrderType.BID.equals(limitOrder.getType())? BUY.toUpperCase() : SELL.toUpperCase())
+                .orderQty(limitOrder.getOriginalAmount())
+                .price(limitOrder.getLimitPrice())
+                .clOrdId(generateClOrdId())
+                .build();
+    }
+
+    private static String generateClOrdId() {
+        String uuid = UUID.randomUUID().toString();
+        uuid = uuid.substring(0, 16).replace("-", "");
+        return uuid;
+    }
+
+    public static FundingRecord.Status toWithdrawStatus(String status) {
+        switch (status.toUpperCase()) {
+            case "REJECTED":
+            case "REFUNDING":
+                return FundingRecord.Status.CANCELLED;
+            case "PENDING":
+                return FundingRecord.Status.PROCESSING;
+            case "FAILED":
+                return FundingRecord.Status.FAILED;
+            case "COMPLETED":
+                return FundingRecord.Status.COMPLETE;
+            default:
+                throw new RuntimeException("Unknown withdraw status: " + status);
+        }
+    }
+    public static FundingRecord.Status toDepositStatus(String status) {
+        switch (status.toUpperCase()) {
+            case "REJECTED":
+            case "UNCONFIRMED":
```

Comment:
https://github.com/blockchain/XChange/blob/develop/xchange-core/src/main/java/org/knowm/xchange/dto/account/FundingRecord.java#L346

It seems that UNCONFIRMED should be mapped to PROCESSING. Please check

Reply:
On the BCDC side we have these 3 states in deposit: REJECTED, UNCONFIRMED, COMPLETED that must be equated with the library states which are: 

 ```
PROCESSING(
        "WAIT CONFIRMATION",
        "EMAIL CONFIRMATION",
        "EMAIL SENT",
        "AWAITING APPROVAL",
        "VERIFYING",
        "PENDING_APPROVAL",
        "PENDING"),
COMPLETE("COMPLETED"),
CANCELLED("REVOKED", "CANCEL", "REFUND"),
 FAILED("FAILURE"),
```

UNCONFIRMED seemed to me to fit better with REVOKED which is inside CANCELLED.


Translated with www.DeepL.com/Translator (free version)